File: code360/26-best_time_to_buy_and_sell_stock.cpp
#include <bits/stdc++.h> 
int maximumProfit(vector<int> &prices){
    // Write your code here.
    int minimumPrice = prices[0];
    int maxProfit = 0;

    for(int i = 0; i < prices.size(); i++){
        int cost = prices[i] - minimumPrice;
        maxProfit = std::max(maxProfit, cost);
        minimumPrice = std::min(minimumPrice, prices[i]);
    }
    return maxProfit;
}

File: code360/26-best_time_to_buy_and_sell_stock.java
import java.util.* ;
import java.io.*; 
import java.util.ArrayList;

public class Solution{
    public static int maximumProfit(ArrayList<Integer> prices){
        // Write your code here.
        int minimumPrice = prices.get(0);
        int maxProfit = 0;
        
        for(int i = 0; i < prices.size(); i++){
            int cost = prices.get(i) - minimumPrice;
            maxProfit = Math.max(maxProfit, cost);
            minimumPrice = Math.min(minimumPrice, prices.get(i));
        }
        return maxProfit;
    }
}